Shared/Sources/DynamicColourViewController.swift

```diff
@@ -0,0 +1,55 @@
+// Copyright © 2020 Itty Bitty Apps. All rights reserved.
+
+import Foundation
+
+final class DynamicColourViewController: RevertTableViewController {
+
+  private let collection: [ColourSection] = RevertItems.coloursData
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+  }
+
+}
+
+extension DynamicColourViewController {
+
+  override func numberOfSections(in tableView: UITableView) -> Int {
+      collection.count
+  }
+  override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+      collection[section].rows.count
+  }
+
+  override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+      collection[section].title
+  }
+
+  override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    guard let cell = tableView.dequeueReusableCell(withIdentifier: "TableViewControllerCell") else {
+      fatalError("Unknown cellIdentifier or cell")
+    }
+
+    let section = collection[indexPath.section]
+    let row = section.rows[indexPath.row]
+
+    if #available(iOS 10.0, *) {
+      let imageSize = CGSize(width: 50, height: 50)
```

Comment:
It's okay to do some cell setup under `cellForRowAt` func. But I think it will be much better if you can just create a new custom class, specifically for setting up Dynamic Colour cells, that inherits from `BasicCell`.

The class should set up the image size, corner radius, label font, background colors, and etc. The default settings basically.
Then it should have a public method or vars that can set up strings to display on the label and system color for `imageView`.

That way we can have some separation of concerns. Under  `tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath)` method, we only care about setting up the cell's content according to the indexpath.

And under the new custom class, we only care about the default cell settings that will be consistently displayed on all cells in the tableview.

Reply:
Correction, the custom class does not need to inherit from `BasicCell`. I just look at the source code and find out which cell class that you are using for this table view.